### PR TITLE
HttpConnectionManager and upstream counters for total completed requests

### PR DESCRIPTION
**Pull request by jsedgwick**
*Description*: add counters for total completed requests regardless of status code
*Risk Level*: low
*Testing*: added relevant in unit tests
*Docs Changes*: none
*Release Notes*: added one
cc @btc

### CODE REVIEW

File: docs/root/configuration/http_conn_man/stats.rst [hunk 1]
```diff
@@ -45,6 +45,7 @@ statistics:
    downstream_rq_tx_reset, Counter, Total request resets sent
    downstream_rq_non_relative_path, Counter, Total requests with a non-relative HTTP path
    downstream_rq_too_large, Counter, Total requests resulting in a 413 due to buffering an overly large body
+   downstream_rq_completed, Counter, Total responses
```
Comment (ccaraman) [inline docs/root/configuration/http_conn_man/stats.rst]: What does completed mean? Can you be more specific about how it differs from _rq_total

**Comment by jsedgwick** [comment]
I'm going to add equivalent counters for `upstream_rq_completed` per cluster to this review.
My VM is acting up but I should have that update in tomorrow. Don't bother reviewing until then.

File: docs/root/configuration/http_conn_man/stats.rst (hunk 1 above)
Comment (jsedgwick) [reply docs/root/configuration/http_conn_man/stats.rst]: total is incremented on request initiation, regardless of response code (if any), whereas this is incremented on request completion
@btc and @danielhochman have the context here in terms of why they wanted it.
i can be more specific in the docs, ya, willfix

**Comment by jsedgwick** [comment]
@danielhochman

**Pull request merged**